### DISCUSSION TITLE
Docs: Extracting a form object

### DISCUSSION THREAD

**Discussion by eliezerben** [livewire/livewire#8158] [1. Help] [answered]
I'm seeing an issue in the last code snippet in this section of the docs. https://livewire.laravel.com/docs/forms#extracting-a-form-object

Here's the full example.
```php
<?php
 
namespace App\Livewire\Forms;
 
use Livewire\Attributes\Validate;
use Livewire\Form;
use App\Models\Post;
 
class PostForm extends Form
{
    public ?Post $post;
 
    #[Validate('required|min:5')]
    public $title = '';
 
    #[Validate('required|min:5')]
    public $content = '';
 
    public function setPost(Post $post)
    {
        $this->post = $post;
 
        $this->title = $post->title;
 
        $this->content = $post->content;
    }
 
    public function store()
    {
        $this->validate();
 
        Post::create($this->only(['title', 'content']));
    }
 
    public function update()
    {
        $this->post->update(
            $this->all()
        );
    }
}
```

This part of the code is giving an error because `$this->all()` also returns the `$post` object.
```php
        $this->post->update(
            $this->all()
        );
```

Is this an error in the docs?

Is there a way to specify that `$post` should not be included in `all()`? Or should I use `$this->only(['title', 'content'])`

**Comment by danie-ramdhani** [comment]
works fine for me?

i know it's missing the `$this->validate();` in PostForm update function (i'll make PR later).

what is the error message?

**Reply by eliezerben** [reply]
This is the error:
```
Illuminate\Database\QueryException
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'post' in 'field list'
update `posts` set `title` = test title, `body` = test body, `post` = ?, `posts`.`updated_at` = 2024-03-19 09:12:13 where `id` = 59
```

This is my code:
```php
<?php

namespace App\Livewire\Forms;

use App\Models\Post;
use Livewire\Attributes\Validate;
use Livewire\Form;

class PostForm extends Form
{

    public ?Post $post;

    #[Validate(['required', 'min:5'])]
    public $title = '';

    #[Validate(['required', 'min:5'])]
    public $body = '';

    public function setPost(Post $post)
    {
        $this->post = $post;
        $this->title = $post->title;
        $this->body = $post->body;
    }

    public function store()
    {
        $this->validate();
        Post::create($this->only(['title', 'body']));
    }

    public function update()
    {
        $this->validate();
        $this->post->update($this->all());
    }
}
```

From what I understand, it is also trying to save the `$post` to the table which it shouldn't.

**Reply by danie-ramdhani** [reply, marked as answer]
idk why i didn't get the error (maybe because i use sqlite (?))

Mmmm... make sense. why we must pass all the data... this one is more laravelish right?

```php
   public function store()
    {
        Post::create($this->validate());
    }

    public function update()
    {
        $this->post->update($this->validate());
    }
```

**Reply by eliezerben** [reply]
`$this->validate()` looks easier to use. Thanks for the idea👍

**Reply by eliezerben** [reply]
BTW, I tried with sqlite, I get the same issue when using `$this->all()`

**Comment by IsuruSankhajith** [comment]
Yes, you're correct in noticing that `$this->all()` includes all the properties of the Livewire form object, including `$post`, which you don't want to include in the update operation. 

To avoid including `$post` in the `update()` method, you can use the `$this->only(['title', 'content'])` method, which explicitly specifies the fields you want to include. This is the preferred approach in this scenario.

So your `update()` method should be modified as follows:

```php
public function update()
{
    $this->post->update(
        $this->only(['title', 'content'])
    );
}
```

Using `$this->only(['title', 'content'])` ensures that only the `title` and `content` fields are passed to the `update()` method of the `Post` model.

Regarding the documentation, it might be an oversight or it might be assuming that you are aware that `$this->all()` includes all properties, including the `$post` object. Regardless, using `$this->only(['title', 'content'])` is the correct and explicit way to specify which fields should be included in the update operation.

**Reply by joshhanley** [reply]
Another option is to use `$this->except(['post']);` if you don't want to have to specify every property and just want to exclude the post property.

Yeah docs might be wrong, PRs to fix are welcome! 🙂

**Reply by eliezerben** [reply]
Thanks @IsuruSankhajith and @joshhanley

**Reply by eliezerben** [reply]
@joshhanley
> $this->except(['post']);

Is this better than using data returned from `$this->validate()` ?

**Reply by joshhanley** [reply]
@eliezerben no worries! 🙂 `validate()` only returns validated data, so it's the safest to use. But if you have already validated all the data on the form then it's fine to use `except()`, the only catch is if you add new properties in the future, using except will mean you need to ensure all new properties are also validated.

**Reply by danie-ramdhani** [reply]
made PR https://github.com/livewire/livewire/pull/8162
